### ISSUE TITLE
Bypassing approval process and registering when site is offline

### ISSUE THREAD

**Issue by kanalumaddela** [Lavoaster/IP.Board-Steam-Authentication-Method#60]
When a new user registers with steam sign, they are able to:
- Bypass approval process (email confirmation + manual admin approval)
- Does not display in the member lists in the admin cp, but can be search for

And on top of that, while both **registration is disabled** through admin cp and the **site is set as offline, the user is able to register** which is a big issue in my opinion. The only way I was a able to disable new members from registering was to disable the login handler

IPS Community Suite v4.1.15

**Comment by Lavoaster** [comment]
@kanalumaddela Can you verify if the registration issue persists within other login handlers?

**Comment by madman32395** [comment]
>Not display in the member lists in the admin cp, but can be search for
>>Thats normal, as per IPS. The reason why is the user doesnt display immediately display is because of lack of email. Once an email is entered, they will display without having to search. so basically the user is halfway registering and stopping.

>Bypass approval process (email confirmation + manual admin approval)
>> @Lavoaster I can confirm that Google does the email confirmation. Also tested Email confirmation with Steam login on another Client's site of mine. and It does indeed ask for email validation (and if applicable, the admin's approval as well)

**Comment by kanalumaddela** [comment]
Apologies for the last response

> Thats normal, as per IPS. The reason why is the user doesn't display immediately display is because of lack of email. 
Thanks for that information, I wasn't aware of it and will keep an eye on that.

The issue with bypassing admin approval was a misinterpretation of how I saw it in the admin panel, but here's why:

I have both google and steam auth enabled. When the site is set as offline, using either of those methods still allows a user to register. Using the google method gets the user "farther" in the registration process and shows the user in the awaiting approval list, but steam sign in doesn't necessarily fully register the user because as madman said the email isn't there, but gets to the point where it takes up a user id which may seem like nothing but can become spam if this were to happen multiple times

**Basically what I'm trying to say is that this seems more of an IPB4 issue where external login handlers allow a user to register while a site is offline.**

I do not know if there's anything that can be done to prevent registration like this on a offline site, but if it is possible, it would be nice.

**Comment by madman32395** [comment]
Yes, they can register. That is normal operation. But until they are given
permissions of an account, they will either be authorized to log in while
offline or unauthorized.

As for member ID spamming. I wouldn't sweat it lol. The system will only
allow them to register once per handler unless they cancel registration.
